compiler/cpp/CMakeLists.txt

```diff
@@ -18,6 +18,16 @@
 #
 
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/src/thrift/version.h.in ${CMAKE_CURRENT_BINARY_DIR}/thrift/version.h)
+if(MSVC)
+    # The winflexbison generator outputs some macros that conflict with the Visual Studio 2010 copy of stdint.h
+    # This might be fixed in later versions of Visual Studio, but an easy solution is to include stdint.h first
+    if(HAVE_STDINT_H)
+        add_definitions(/FI"stdint.h")
+    endif(HAVE_STDINT_H)
+
+    # The generated code also causes a size type warning we cannot work around easily
+    set_source_files_properties(thriftl.cc PROPERTIES COMPILE_FLAGS /wd4267)
```

Comment:
I was able to move this into thriftc.ll however we still need the /FIstdint.h to avoid compiler warnings.